Doc/library/sys.rst

```diff
@@ -1282,6 +1282,59 @@ always available. Unless explicitly noted otherwise, all variables are read-only
 
    .. versionadded:: 3.5
 
+.. data:: _jit
+
+   Utilities for observing just-in-time compilation.
+
+   .. impl-detail::
+
+      JIT compilation is an *experimental implementation detail* of CPython.
+      ``sys._jit`` is not guaranteed to exist or behave the same way in all
+      Python implementations, versions, or build configurations.
+
+   .. versionadded:: 3.14
+
+   .. function:: sys._jit.is_available()
+
+      Return ``True`` if the current Python executable supports JIT compilation,
+      and ``False`` otherwise.  This can be controlled by building CPython with
+      the ``--experimental-jit`` option on Windows, and the
+      :option:`--enable-experimental-jit` option on all other platforms.
+
+   .. function:: sys._jit.is_enabled()
+
+      Return ``True`` if JIT compilation is enabled for the current Python
+      process (implies ``sys._jit.is_available()``), and ``False`` otherwise.
+      If JIT compilation is available, this can be controlled by setting the
+      :envvar:`PYTHON_JIT` environment variable to ``0`` (disabled) or ``1``
+      (enabled) at interpreter startup.
+
+   .. function:: sys._jit.is_active()
+
+      Return ``True`` if the topmost Python frame is currently executing JIT
+      code, and ``False`` otherwise.
+
+      .. note::
+
+         Due to the nature of tracing JIT compilers, repeated calls to this
+         function may give surprising results. For example, branching on its
+         return value will likely lead to unexpected behavior (if doing so
+         causes JIT code to be entered or exited):
```

Comment:
I think this section should have more emphasis that this is mainly for testing/debugging, per your rationale in Discourse:

> * It seems useful for tests for the JIT itself.
> * Lots of people have asked: “How do I know if the JIT is actually working?”

I would also suggest upgrading the 'note' about branching to a 'warning', if nothing else changes.

Reply:
I had thought that "warning" was reserved for security issues, or possible misuse of an API that could corrupt interpreter state or lead to crashes. This is just saying "the results may not be what you expect". Is a warning more appropriate than a note for calling out quirks like this?

Reply:
How's adding this to the top of the note:

> This function is intended for testing and debugging the JIT itself. Beyond using it to confirm that the JIT is actually working, most users will never *need* to use it.

Reply:
> I had thought that "warning" was reserved for security issues, or possible misuse of an API that could corrupt interpreter state or lead to crashes.

That would be "caution" or "danger"! The full [list of admonitions](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#admonitions-messages-and-warnings) is tip, hint, note, important, attention, warning, caution, error, and danger.

If you think "warning" is too much though, perhaps "attention" or "important".

The suggested text reads well, an alternative proposal would be:

> This function is only intended for testing and debugging the implementation of the JIT. It should be avoided for any other use-cases.

Reply:
This is the more pragmatic approach, sounds good!